apisix/init.lua

```diff
@@ -297,6 +300,32 @@ local function return_direct(...)
 end
 
 
+local function set_upstream_host(api_ctx)
+    local pass_host = api_ctx.pass_host
+    if pass_host and pass_host ~= "pass" then
```

Comment:
How about check pass_host directly? Because there has no logics to handle the case when pass_host doesn’t exist 🤔

```lua
local host

If pass_host == “xx” then
   xxx

If pass_host == “xx” then
  xxx
```